pkg/network/dhcp_daemon.go

```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"encoding/json"
+	operv1 "github.com/openshift/api/operator/v1"
+	"github.com/pkg/errors"
+	"log"
+)
+
+// UseDHCP determines if the the DHCP CNI plugin running as a daemon should be rendered.
+func UseDHCP(conf *operv1.NetworkSpec) (bool, error) {
+	renderdhcp := false
+
+	// This isn't useful without Multinetwork.
+	if *conf.DisableMultiNetwork {
+		return renderdhcp, nil
+	}
+
+	// Look and see if we have an AdditionalNetworks
+	if conf.AdditionalNetworks != nil {
+		for _, addnet := range conf.AdditionalNetworks {
+			// Parse the RawCNIConfig
+			var rawConfig map[string]interface{}
+			var err error
+
+			confBytes := []byte(addnet.RawCNIConfig)
+			err = json.Unmarshal(confBytes, &rawConfig)
+			if err != nil {
+				log.Printf("WARNING: Not rendering DHCP daemonset, failed to Unmarshal RawCNIConfig: %v", confBytes)
+				return renderdhcp, nil
+			}
+
+			// Cycle through the IPAM keys, and determine if the type is dhcp
+			if rawConfig["ipam"] != nil {
+				ipam, ok := rawConfig["ipam"].(map[string]interface{})
+				if !ok {
+					return renderdhcp, errors.Errorf("IPAM element has data of type %T but wanted map[string]interface{}", rawConfig["ipam"])
+				}
+
+				for key, value := range ipam {
+					if key == "type" {
+						if value.(string) == "dhcp" {
```

Comment:
The `for key, value` doesn't help here; you're still doing an unsafe cast. (eg,  it would crash on `{"ipam": { "type": 5 }}`

Reply:
Great catch, missed that one, thanks.